src/ucp/wireup/wireup.c

```diff
@@ -718,6 +718,11 @@ ucp_wireup_send_ep_removed(ucp_worker_h worker, const ucp_wireup_msg_t *msg,
         goto out_cleanup_lanes;
     }
 
+    /* Set UCP_EP_FLAG_USED flag to make sure that the UCP EP won't be deleted
+     * upon detecting and handling peer failure which destroyes UCP EPs which
+     * don't have UCP_EP_FLAG_USED flag set (i.e. UCP EPs were not exposed to
+     * a user or used as reply EP for sending WIREUP/EP_REMOVED message) */
+    ucp_ep_update_flags(reply_ep, UCP_EP_FLAG_USED, 0);
```

Comment:
can we additionally set `UCP_EP_FLAG_INTERNAL` to diverse internal/external usage? IMO this is important for debugging

Reply:
maybe we even can avoid reusing `UCP_EP_FLAG_USED` in this case, because currently it represents `external` usage

Reply:
we can't set `UCP_EP_FLAG_INTERNAL`, it won't be deleted from UCP Worker's EP list here:
https://github.com/openucx/ucx/blob/01b4d70b4f01c539dee0a87b03d150b34da50fca/src/ucp/core/ucp_ep.c#L350

Reply:
> we can't set `UCP_EP_FLAG_INTERNAL`, it won't be deleted from UCP Worker's EP list here:
> https://github.com/openucx/ucx/blob/01b4d70b4f01c539dee0a87b03d150b34da50fca/src/ucp/core/ucp_ep.c#L350

then we have to handle `USED | INTERNAL` as a separate case, it's just clearer

Reply:
ep_list is what we need to flush/keepalive, and we probably don't need to flush/ka that ep
IMO, we should make new list on the worker for internal ep's. and for WIREUP/EP_REMOVED, set only INTERNAL flag, and not set USED flag.